src/test/java/cz/cvut/kbss/termit/rest/ResourceControllerTest.java

```diff
@@ -482,6 +491,34 @@ void addFileToDocumentReturnsConflictWhenRequestedResourceIsNotDocument() throws
                .andExpect(status().isConflict());
     }
 
+    @Test
+    void removeFileFromDocumentReturnsOkForAValidFile() throws Exception {
+        final Resource resource = Generator.generateResource();
+        resource.setUri(RESOURCE_URI);
+        resource.setLabel(RESOURCE_NAME);
+        final File fOne = generateFile();
+        when(identifierResolverMock.resolveIdentifier(RESOURCE_NAMESPACE, RESOURCE_NAME)).thenReturn(RESOURCE_URI);
+        when(resourceServiceMock.findRequired(RESOURCE_URI)).thenReturn(resource);
+        doThrow(UnsupportedAssetOperationException.class).when(resourceServiceMock).addFileToDocument(resource, fOne);
+        mockMvc.perform(post(PATH + "/" + RESOURCE_NAME + "/files").param(QueryParams.NAMESPACE, RESOURCE_NAMESPACE)
+            .content(toJson(fOne))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isConflict());
+
+        when(resourceServiceMock.findRequired(RESOURCE_URI)).thenReturn(resource);
+        mockMvc
+            .perform(delete(PATH + "/" + RESOURCE_NAME + "/files/" + FILE_NAME, RESOURCE_NAMESPACE))
+            .andExpect(status().isNoContent());
```

Comment:
This test seems to be doing two things. First attempting to add a file to a document and them remove it. The first part does not make sense to me, given the test name.